src/aggregate_verify.cc

```diff
@@ -0,0 +1,226 @@
+    uint8_t *msg;
+    size_t msg_len;
+} AggregateVerifySet;
+
+/**
+ * @param[out]  {blst_ts::P2AffineGroup} - P2 point for verification
+ * @param[out]  {std::vector<AggregateVerifySet>} - Sets to be added to pairing
+ * @param[in]   {Napi::CallbackInfo} - JS function context
+ */
+blst_ts::BLST_TS_ERROR prepare_aggregate_verify(
+    blst_ts::P2AffineGroup &sig_point,
+    std::vector<AggregateVerifySet> &sets,
+    const Napi::CallbackInfo &info) {
+    Napi::Env env = info.Env();
+    blst_ts::BLST_TS_ERROR err = unwrap_signature(sig_point, env, info[2]);
+    if (err != blst_ts::BLST_TS_ERROR::SUCCESS) {
+        return err;
+    }
+
+    if (!info[0].IsArray()) {
+        Napi::TypeError::New(
+            env, "BLST_ERROR: msgs must be of type BlstBuffer[]")
+            .ThrowAsJavaScriptException();
+        return blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN;
+    }
+    Napi::Array msgs_array = info[0].As<Napi::Array>();
+    uint32_t msgs_array_length = msgs_array.Length();
+    if (msgs_array_length == 0) {
+        Napi::TypeError::New(env, "BLST_ERROR: msgs must have length > 0")
+            .ThrowAsJavaScriptException();
+        return blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN;
+    }
+
+    if (!info[1].IsArray()) {
+        Napi::TypeError::New(env, "publicKeys must be of type PublicKeyArg[]")
+            .ThrowAsJavaScriptException();
+        return blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN;
+    }
+    Napi::Array pk_array = info[1].As<Napi::Array>();
+    uint32_t pk_array_length = pk_array.Length();
+    if (pk_array_length == 0) {
+        if (sig_point.raw_point->is_inf()) {
+            return blst_ts::BLST_TS_ERROR::INVALID;
+        }
+        Napi::TypeError::New(env, "BLST_ERROR: publicKeys must have length > 0")
+            .ThrowAsJavaScriptException();
+        return blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN;
+    }
+
+    if (msgs_array_length != pk_array_length) {
+        Napi::TypeError::New(
+            env, "BLST_ERROR: msgs and publicKeys must be the same length")
+            .ThrowAsJavaScriptException();
+        return blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN;
+    }
+
+    sets.reserve(pk_array_length);
+    for (uint32_t i = 0; i < pk_array_length; i++) {
+        sets.push_back({blst_ts::P1AffineGroup{}, nullptr, 0});
+
+        Napi::Value msg_value = msgs_array[i];
+        BLST_TS_FUNCTION_UNWRAP_UINT_8_ARRAY(msg_value, msg, "msg");
+        sets[i].msg = msg.Data();
+        sets[i].msg_len = msg.ByteLength();
+
+        err = unwrap_public_key(sets[i].pk_point, env, pk_array[i]);
+        if (err != blst_ts::BLST_TS_ERROR::SUCCESS) {
+            return err;
+        }
+    }
+
+    return blst_ts::BLST_TS_ERROR::SUCCESS;
+}
+
+/**
+ * @param[out]  {boolean} - Result of the verification
+ * @param[out]  {std::string} - Error message for invalid aggregate
+ * @param[in]   {BlstTsAddon *} - Addon module
+ * @param[in]   {blst::Paring *} - Pointer to paring for the verification
```

Comment:
Is there a reason you're duplicating the type in the documentation? Not sure what documentation software are you using, but most should easily deduce it.

Reply:
Trying to make as many comments/docs for future maintainers as possible.  Not sure how to handle in/out params from a documentation perspective other than docstrings like these.  Is there a better way?

Reply:
It would be great if the params didn't require documentation (self-documenting), except for a high-level description of the function. I'd follow how it's done in Rust, e.g., https://doc.rust-lang.org/src/std/collections/hash/map.rs.html#688. Limiting the number of parameters and not using `out` ones would likely help to achieve that.

The problem I see with such params documentation is that it introduces additional maintenance burden - say, a type has changed. Refactoring tools will modify the code but likely won't touch the documentation - and wrong documentation is even worse than no documentation at all! :D For example, in the doc you say:

```
 * @param[in]   {blst::Paring *} - Pointer to paring for the verification
```

But the code says
```
std::unique_ptr<blst::Pairing> &ctx,
```

Reply:
Oy jeez... Good call.  I'll just go with the param name and the types can be inferred from intellisense.